holosqape/socketinterface.cpp

```diff
@@ -0,0 +1,111 @@
+#include "socketinterface.h"
+#include <iostream>
+#include <QJsonDocument>
+#include "app.h"
+
+SocketInterface::SocketInterface(QObject *parent) : QObject(parent),
+    m_server(QString("Holochain Container"), QWebSocketServer::NonSecureMode, this),
+    m_container(0)
+{
+    //("192.168.1.100")
+    if (m_server.listen(QHostAddress("192.168.1.100"), 8888)) {
+        std::cout << "Listening on port 8888" << std::endl;
+    } else {
+        std::cout << "Could not listen on port!" << std::endl;
+    }
+    QObject::connect(&m_server, SIGNAL(newConnection()), this, SLOT(incomingConnection()));
+    QObject::connect(&m_server, SIGNAL(serverError(QWebSocketProtocol::CloseCode)), this, SLOT(serverError(QWebSocketProtocol::CloseCode)));
+
+}
+
+void SocketInterface::setContainer(Container* container) {
+    m_container = container;
+}
+
+
+void SocketInterface::incomingConnection() {
+    std::cout << "Incoming connection.." << std::endl;
+    QWebSocket* socket = m_server.nextPendingConnection();
+    if(!socket) return;
+
+    std::cout << "Socket good" << std::endl;
+
+    m_web_sockets << socket;
+    QObject::connect(socket, &QWebSocket::textMessageReceived,
+                     this, &SocketInterface::message_received
+                     );
+}
+
+void SocketInterface::message_received(const QString &message) {
+    std::cout << "Message: ";
+    std::cout << message.toStdString() << std::endl;
+
+    QWebSocket* socket = qobject_cast<QWebSocket*>(sender());
+
+
+    QJsonDocument doc = QJsonDocument::fromJson(message.toUtf8());
+
+    if(!doc.isObject()) {
+        std::cout << "not an object" << std::endl;
+        socket->sendTextMessage(QString('{"error": "recevied non-JSON-object"}'));
```

Comment:
typo

Reply:
this could be not right for this case, but in the core dev team chats we've talked about being willing/open to just push commits with typo fixes :)